udf/agent/py/kapacitor/udf/agent.py

```diff
@@ -68,6 +85,19 @@ def end_batch(self, end_req):
 # The Agent requires a Handler object in order to fulfill requests.
 class Agent(object):
     def __init__(self, _in=defaultIn, out=defaultOut,handler=None):
+        if is_python_2:
+            import datetime
+            now_utc = datetime.datetime.utcnow()
+            lgr = logging.getLogger("DEPRECATION")
+            dpr_handler = logging.FileHandler("DEPRECATION_WARNING_PYTHON_2.txt")
+            lgr.addHandler(dpr_handler)
+            lgr.setLevel(logging.WARNING)
+            frame = inspect.stack()[1]
+            lgr.warning("[DEPRECATION WARNING] - %s - detected python2 in %s  "
+                        "Support for Python 2-based UDFs is deprecated as of Kapacitor 1.7.7 and "
+                        "will be removed in Kapacitor 1.8.0. Please update your UDFs to be "
+                        "Python 3-compatible before upgrading. This change is part of our effort to "
+                        "follow modern security best practices.", now_utc, frame[0].f_code.co_filename)
```

Comment:
Please remove this. Keep Agent as is for better maintainability and to ensure we don’t accidentally break something.